### PR TITLE
fix(lint): fix linting error

### PR DESCRIPTION
**Pull request by scortier**
- Add `golangci-lint` to the workflow
- Add `lint` command 
- Resolve linting errors

### CODE REVIEW

**Comment by scortier** [comment]
Related Issues: #38 #39

File: lineage/graph.go [hunk 1]
```diff
@@ -59,7 +57,7 @@ func (graph InMemoryGraph) Query(cfg QueryCfg) (AdjacencyMap, error) {
 		var err error
 		supergraph, err = graph.buildSubgraphFromRoot(supergraph, cfg.Root)
 		if err != nil {
-			return supergraph, errors.Wrap(err, "error building subgraph")
+			return supergraph, fmt.Errorf("error building subgraph %s", err)
```
Comment (mabdh) [inline lineage/graph.go:60]:
```suggestion
			return supergraph, fmt.Errorf("error building subgraph: %w", err)
```
If it is error wrapping, we could use `%w`.
see https://go.dev/blog/go1.13-errors

Comment (scortier) [reply lineage/graph.go:60]: Should we replace all error wrapping with `fmt.Errorf` as `errors.wrap` package has been archived.
wdyt ??

Comment (mabdh) [reply lineage/graph.go:60]: yes let's do it. I guess that should be no problem, we could do that as part of this story.

Comment (scortier) [reply lineage/graph.go:60]: sounds good

**Comment by mabdh** [comment]
@scortier I still see `github.com/pkg/errors` package in `go.mod`. Have you tried doing `go mod tidy`?

**Comment by scortier** [comment]
> @scortier I still see `github.com/pkg/errors` package in `go.mod`. Have you tried doing `go mod tidy`?

Yes, I did

**Review by mabdh — APPROVED** [review]

**Pull request merged**